### ISSUE TITLE
Add preference: disable mouse hover on the thumbnails

### ISSUE THREAD

**Issue by janbl** [lwouis/alt-tab-macos#125]
Very often I want a fast app switch when I parallel move the mouse. 
At the moment the mouse move selects a random app. 
I would like to disable this feature and switch on mouse click only or make this a preference.

**Comment by lwouis** [comment]
Hey @janbl, thanks for taking time to share your experience!

Since #34, the user can hover the thumbnails to change the selected window. Then, either on release of the shortcut, or on mouse click, the window will be selected.

I understand from you that sometimes you're moving the mouse while you trigger alt-tab, and the unintentional hovering of the thumbnails messes up your window selection, and once you release, it selects the wrong one.

Because people clearly want mouse control, it seems like a preference is the way to go. Adding it would be less than 10 lines of code I believe. We have checkbox already in the preferences, so it's about copying that code, then it's about disabling the hovering if it's off

**Comment by janbl** [comment]
Hi @lwouis, that sounds like the ideal solution and would make it far more usable for me.

**Comment by lwouis** [comment]
@janbl I forgot that I had actually plans for this in #72. You can see there how the shortcut/mouse preferences UI would look like.

Note that the "Select windows with mouse" phrasing is ambiguous. It doesn't separate mouse hover from mouse click. However, no-one has requested to disable mouse click so far, so for the time being I think having this UI disable hover interaction would be reasonable.

Preferences are always a balancing act between giving more control to power-users, and not overwhelming casual-users going there to change very basic functionality.

**Comment by PLanB2008** [comment]
+1 for this. Using the mouse while pressing altTab always selects a window I don't really want. In my opinion its more about the selecting the window by hovering than actual selecting it by mouseklick.
Having the option to disable the mousesupport all together would be a great improvement :)

**Comment by lwouis** [comment]
@PLanB2008 please upvote https://github.com/Kentzo/ShortcutRecorder/issues/114 if you want #72 to move forward. I'm waiting on this to start work on #72

**Comment by mfn** [comment]
I've also been "hit" by times at times.

HyperSwitch solves this, don't know technically how, in that the first few pixels? ms? of movementsare in fact **not** triggering the select of the underlying window.

But when you move your mouse more, it does.

**Comment by lwouis** [comment]
> HyperSwitch solves this, don't know technically how, in that the first few pixels? ms? of movementsare in fact **not** triggering the select of the underlying window.

It does indeed have some interesting behavior. I tested and it's definitely not time based. It is distance based. I think they ignore the first 50px of mouse movement over the whole window. After 50px, they let the mouse interact.

It is not enough for some used like @PLanB2008 who just want mouse off all together, but it may be interesting to implement *in addition*. Thank you for sharing this observation @mfn!

**Comment by lwouis** [comment]
I just released v3.3.0 which contains a new preference to disable mouse hover selection. I hope you guys enjoy! 👍

Feel free to open a ticket if you have any issue with the new version

**Comment by janbl** [comment]
Hi,

can I send you my crash report of alttab 3.3.0

Chears,

Jan

**Comment by lwouis** [comment]
@janbl i think you are having the same crash as in #202. Could you please see my reply there?

**Comment by janbl** [comment]
defaults read com.lwouis.alt-tab-macos

{

    "NSWindow Frame SUUpdateAlert" = "1410 557 700 622 0 0 3440 1417 ";

    SUHasLaunchedBefore = 1;

    SULastCheckTime = "2020-04-02 11:17:23 +0000";

    SUUpdateRelaunchingMarker = 1;

    alignThumbnails = Center;

    metaKey = "\\U2318 command";

    startAtLogin = true;

    tabKeyCode = 48;

    theme = "\\Uf8ff macOS";

    windowDisplayDelay = 0;

}

Jan

**Comment by lwouis** [comment]
Let's continue the conversation in the new ticket #202 ;)
